src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/abstractItems/AGenerator.java

```diff
@@ -297,7 +296,7 @@ public double generateEnergy(Location l, SlimefunItem sf, Config data) {
 						processing.put(l, r);
 						progress.put(l, r.getTicks());
 					}
-					return 0;
+					return ChargableBlock.getCharge(l);
```

Comment:
This should check "ChargableBlock.isChargable(l)" before-hand, otherwise we can run into issues.